cfme/common/provider.py

```diff
@@ -409,8 +410,8 @@ def wait_for_delete(self):
     def _on_detail_page(self):
         """ Returns ``True`` if on the providers detail page, ``False`` if not."""
         ensure_browser_open()
-        return sel.is_displayed(
-            '//div[@class="dhtmlxInfoBarLabel-2"][contains(., "{} (Summary)")]'.format(self.name))
+        title = '{} (Summary)'.format(self.name)
+        return breadcrumbs() == ['Infrastructure Providers', title] and summary_title() == title
```

Comment:
Doh, good point. I will fix it up to be more common once I get a run with this commit.